src/components/chart/vegaLiteChart/vegaLiteChart.tsx

```diff
@@ -0,0 +1,39 @@
+import './vegaLiteChart.css'
+
+import Typography from '@mui/material/Typography'
+import React, { useEffect, useRef, useState } from 'react'
+import { default as embed } from 'vega-embed'
+
+import type { VegaLiteChartData } from '../../types'
+
+/** The `VegaLiteChart` component is a versatile tool for visualizing data using the Vega-Lite grammar. With support for various chart types, it empowers users to create engaging and informative data visualizations effortlessly. */
+function VegaLiteChart(props: VegaLiteChartData) {
+  const chartRef = useRef<HTMLDivElement>(null)
+  const [hasError, setHasError] = useState<boolean>(false)
+
+  useEffect(() => {
+    if (chartRef.current && props.spec) {
+      embed(chartRef.current, props.spec)
+        .then(() => {
+          hasError && setHasError(false)
+        })
+        .catch(() => {
+          setHasError(true)
+        })
+    }
+  }, [])
+
+  if (hasError) {
+    return <Typography variant="body2">Failed to load the chart.</Typography>
```

Comment:
Are there other ways of triggering the error? In storybook you use "invalid chart"- do we want to recommend to check the spec or schema?

Reply:
Can't think of other error cases. I thought about mentioning invalid spec in the error message, but end users might not have any controls for that. What do you think?

Reply:
End users might not be able to control directly but I'm thinking maybe it might help them try to refine their prompt?

Reply:
I dont think we should worry too much about this. This error message is ok